git_pki/gpg_wrapper.py

```diff
@@ -38,25 +39,30 @@ def import_public_key(self, armored):
             print(f"Importing {armored}")
         return self.gpg.import_keys(armored).results
 
-    def private_keys_list(self):
+    def private_keys_list(self, names=None):
         keys = self.gpg.list_keys(True)
+        keys = filter(lambda key: self.__raw_key_matches(key, names), keys)
         return map(self.parse_key, keys)
 
     def private_key_fingerprint(self, name):
-        keys = self.gpg.list_keys(True, keys=name)
-        return keys[0]["fingerprint"].lower() if keys else None
+        keys = self.private_keys_list(name)
+        key = next(keys, None)
+        return None if key == None else key.fingerprint
```

Comment:
nit:
```suggestion
        return None if key is None else key.fingerprint
```
and few other places below